src/ext/java/org/opentripplanner/ext/siri/updater/azure/SiriAzureUpdaterParameters.java

```diff
@@ -28,6 +30,22 @@ public String getType() {
     return type;
   }
 
+  public AuthenticationType getAuthenticationType() {
+    return authenticationType;
+  }
+
+  public void setAuthenticationType(AuthenticationType authenticationType) {
+    this.authenticationType = authenticationType;
+  }
```

Comment:
We generally prefer to set the properties in the constructor so the object is immutable. You could even use a record which makes this very easy.

Reply:
I do agree but in this case such approach would not work. We have two separate parameter classes which both extend same abstract class. If we would make fields in those classes final then we would need to duplicate code for mapping.